File: bubble_sort.rb
def bubble_sort(arr)
  k = 0

  while k < (arr.length - 1)
    i = 0
    j = i + 1
    (arr.length - 1).times do
      # puts "i:#{i}, j:#{j}, k:#{k} "
      if arr[i] > arr[j]
        temp = arr[i]
        arr[i] = arr[j]
        arr[j] = temp
      end
      i += 1
      j += 1
    end
    k += 1
  end
  return arr
end

# def swap(arr[i], arr[j])
#   temp = arr[i]
#   arr[i] = arr[j]
#   arr[j] = temp
# end
print bubble_sort([33,14,27,35,10, 4, 5, 1])
puts ""
print bubble_sort([33,14])
puts ""
print bubble_sort([33])

File: bubbleSort.js
var bubbleSort = function(array) {
  var iterationCount = 0

  for (i = 0, l = array.length - 1; i < l; i++) {
    var switchUp = false
    iterationCount += 1
    for (j = 0, k = array.length - 1; j < k; j++) {
      if (array[j] > array[j+1]) {
        var temp = array[j]
        array[j] = array[j+1]
        array[j+1] = temp
        switchUp = true
      }
    }
    if (switchUp === false) {
      console.log(iterationCount)
      return array
    }
  }
  console.log(iterationCount)
  return array
}
// bubbleSort([33,14])
console.log(bubbleSort([1,2,3,4,5]))
console.log("===========")
console.log(bubbleSort([10,33,14,40,27,35,1]))
